src/context/LocationsContext.js

```diff
@@ -29,23 +30,34 @@ const LocationsContextProvider = (props) => {
 
     if (navigator.geolocation && allowLocation) {
       setLoading({ loading: true, message: 'loading' });
-      navigator.geolocation.getCurrentPosition(
-        (position) => {
-          setUserLocation(position.coords);
-          setLoading({ loading: false, message: '' });
-        },
-        (error) => {
-          console.error(error);
-          setLoading({
-            loading: false,
-            message: 'location services turned off',
-          });
-        },
+      setId((id) =>
```

Comment:
I'm a little confused by this function. We're passing in `id` but that variable is available from the enclosing scope, so you could just omit that. But more importantly, it doesn't appear to set the ID, or am I missing something?

This looks more like the function should be named `enableWatch` or something, if I'm understand its purpose?

Reply:
My understanding of the documentation here is that the watchPosition method returns a watch id that you can pass to clear watch to reference the correct position to stop. So here we are trying to set that watchId to state and then call it again in clearWatch on line 50.

The id isn't read on line 33 so I removed that.